src/EbmlBinary.cpp

```diff
@@ -47,15 +47,23 @@ EbmlBinary::EbmlBinary()
 {}
 
 EbmlBinary::EbmlBinary(const EbmlBinary & ElementToClone)
-  :EbmlElement(ElementToClone)
+  :EbmlElement(ElementToClone), Data(nullptr)
 {
+  *this = ElementToClone;
+}
+
+EbmlBinary &
+EbmlBinary::operator=(const EbmlBinary & ElementToClone)
+{
+  free(Data);
   if (ElementToClone.Data == nullptr)
```

Comment:
How does this work with the parent class ? Are fields automatically copied over ? Or we should do something like

```cpp
*EbmlElement::this = EbmlElement::ElementToClone;
```

Also `EbmlString::operator=` calls `SetValueIsSet();` because the receiver may not have this value set before. (it's probably missing in other `operator=` implementations as well). But it fact it should only be set if the source also had it, otherwise we can't just assume the value is set now.

Reply:
> How does this work with the parent class ? Are fields automatically copied over ? Or we should do something like

The ISO C++ FAQ [answers this question](https://isocpp.org/wiki/faq/assignment-operators#assignment-op-in-derived-class) in depth. TLDR: yes, you need to call the base class' assignment operator.

> Also `EbmlString::operator=` calls `SetValueIsSet();` because the receiver may not have this value set before. (it's probably missing in other `operator=` implementations as well). But it fact it should only be set if the source also had it, otherwise we can't just assume the value is set now.

That's a different issue than what this PR does, though. It'd be good to open an issue for it.